Modules/timemodule.c

```diff
@@ -2390,3 +2399,60 @@ pysleep(PyTime_t timeout)
     return -1;
 #endif
 }
+
+
+#ifndef MS_WINDOWS
+// time.sleep(0) optimized implementation.
+// On error, raise an exception and return -1.
+// On success, return 0.
+//
+// Rationale
+// ---------
+// time.sleep(0) accumulates delays in the generic implementation, but we can
+// skip some calls to `PyTime_Monotonic()` and other checks when the timeout
+// is zero. For details, see https://github.com/python/cpython/pull/128274.
+static int
+pysleep_zero_posix(void)
```

Comment:
Can you write a single pysleep_zero() function for all platforms, and have a code path for Windows? Move this code:

```
        Py_BEGIN_ALLOW_THREADS
        // A value of zero causes the thread to relinquish the remainder of its
        // time slice to any other thread that is ready to run. If there are no
        // other threads ready to run, the function returns immediately, and
        // the thread continues execution.
        Sleep(0);
        Py_END_ALLOW_THREADS
```